src/System.Management.Automation/engine/Attributes.cs

```diff
@@ -1354,13 +1357,82 @@ public ValidateCountAttribute(int minLength, int maxLength)
     }
 
     /// <summary>
+    /// Base class for all valid values generators with cache support.
+    /// </summary>
+    public abstract class CachedValidValuesGeneratorBase : IValidateSetValuesGenerator
+    {
+        // Cached valid values.
+        private string[] _validValues;
+
+        /// <summary>
+        /// Sets a time interval in seconds to reset the '_validValues' dynamic valid values cache.
+        /// By default 'ValidValuesCacheExpiration = 0' and no valid values is cached (is re-evaluated every time it is used).
+        /// </summary>
+        public int ValidValuesCacheExpiration { 
```

Comment:
I'm suggesting:

```C#
    class GenValuesWithExpiration : System.Management.Automation.CachedValidValuesGeneratorBase {
        GenValuesWithExpiration() : base(2) {}
        # Rest is the same
```

The PowerShell engine is calling the default constructor, but the abstract base doesn't need a default constructor.